### PR TITLE
 [FLINK-13279][table-sql-client] Fully qualify sink name in sql-client 

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

This PR fixes the problem that sink registered automatically via sql-client is not accessible when the current catalog or database is changed to something different than the built-in ones. 

## Brief change log

See commit messages

## Verifying this change

This change added tests and can be verified as follows:
* testStreamQueryExecutionSink (test extended)

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / b / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / **JavaDocs** / not documented)

### CODE REVIEW

**Comment by dawidwys** [comment]
Could you have a look at this PR @twalthr?

**Comment by xuefuz** [comment]
Thanks to @dawidwys for the fix, which seems good to me. Here I'd like to reemphasize an observation for future consideration.

In our original design of CatalogManager, we wanted it to be only aware of multiple catalogs with one being the current. CatalogManager doesn't need to treat any one of them specially. Each catalog may has its own capability and restriction, but upper layers doesn't need to care about. This design would avoid such pervasive, special treatment of the built-in catalog as we see in this PR.

While it's too late to revisit this, I'd suggest we open up the discussion again post 1.9.

Thanks!

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala [hunk 1]
```diff
@@ -60,9 +60,9 @@ abstract class TableEnvImpl(
     private val catalogManager: CatalogManager)
   extends TableEnvironment {
 
-  // The current catalog and database are definitely builtin.
-  protected val builtinCatalogName: String = catalogManager.getCurrentCatalog
-  protected val builtinDatabaseName: String = catalogManager.getCurrentDatabase
+  protected val builtInCatalogName: String = catalogManager.getBuiltInCatalogName
```
Comment (twalthr) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala]: nit: remove those variables and access catalog manager directly

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/utils/SimpleCatalogFactory.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+/**
+ * Catalog factory for an in-memory catalog that contains a single non-empty table.
+ * The contents of the table are equal to {@link SimpleCatalogFactory#TABLE_CONTENTS}.
+ */
+public class SimpleCatalogFactory implements CatalogFactory {
+
+	public static final String CATALOG_TYPE_VALUE = "simple-catalog";
+
+	public static final String TEST_TABLE_KEY = "test-table";
```
Comment (twalthr) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/utils/SimpleCatalogFactory.java]: nit: `TEST_TABLE_NAME`

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java [hunk 3]
```diff
@@ -151,6 +155,44 @@ public void testValidateSession() throws Exception {
 		assertEquals(expectedTables, actualTables);
 	}
 
+	@Test(timeout = 30_000L)
+	public void testDefaultCatalogDifferentFromBuiltin() throws Exception {
+		Environment defaultEnvironment = new Environment();
+		Map<String, Object> catalogs = new HashMap<>();
+		catalogs.put(CatalogEntry.CATALOG_NAME, "simple-catalog");
```
Comment (twalthr) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java]: add this to the yaml file instead?

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java [hunk 4]
```diff
@@ -151,6 +155,44 @@ public void testValidateSession() throws Exception {
 		assertEquals(expectedTables, actualTables);
 	}
 
+	@Test(timeout = 30_000L)
+	public void testDefaultCatalogDifferentFromBuiltin() throws Exception {
```
Comment (twalthr) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java]: Most IT cases here are testing runtime behavior, I'm not sure if we need to run a full IT case. But I'm fine if there is no better solution yet.

Comment (dawidwys) [reply flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java]: I agree. I also don't like the test. I couldn't find a well isolated way to test this behavior though.

**Pull request closed without merging**